scripts/timelapse-colorizer-data/convert_emt_migration_data.py

```diff
@@ -0,0 +1,215 @@
+    centroids_y = dataset[CENTROIDS_Y_COLUMN].to_numpy()
+
+    feature_data = []
+    for i in range(len(features)):
+        # TODO normalize output range excluding outliers?
+        f = dataset[features[i]].to_numpy()
+        feature_data.append(f)
+
+    writer.write_feature_data(
+        feature_data,
+        tracks,
+        times,
+        centroids_x,
+        centroids_y,
+        outliers,
+    )
+
+
+def make_dataset(
+    data, output_dir="./data/", dataset="3500005820_3", do_frames=True, scale=1
+):
+    """Make a new dataset from the given data, and write the complete dataset
+    files to the given output directory.
+    """
+    writer = ColorizerDatasetWriter(output_dir, dataset, scale=scale)
+    full_dataset = data
+    logging.info("Loaded dataset '" + str(dataset) + "'.")
+
+    # Make a reduced dataframe grouped by time (frame number).
+    columns = [
+        TRACK_ID_COLUMN,
+        TIMES_COLUMN,
+        SEGMENTED_IMAGE_COLUMN,
+        OBJECT_ID_COLUMN,
+    ]
+    reduced_dataset = full_dataset[columns]
+    reduced_dataset = reduced_dataset.reset_index(drop=True)
+    reduced_dataset[INITIAL_INDEX_COLUMN] = reduced_dataset.index.values
+    grouped_frames = reduced_dataset.groupby(TIMES_COLUMN)
+
+    # Make the features, frame data, and manifest.
+    nframes = len(grouped_frames)
+    features = [
+        "mean migration speed per track (um/min)",
+        "Integrated Distance (um)",
+        "Displacement (um)",
+        "Average colony overlap per track",
+        "migration velocity (um/min)",
+        "R0Cell_Neighbors_NumberOfNeighbors_Adjacent",
+        "R0Cell_Neighbors_PercentTouching_Adjacent",
+    ]
+    make_features(full_dataset, features, writer)
+    if do_frames:
+        make_frames(grouped_frames, scale, writer)
+    writer.write_manifest(nframes, features)
+
+
+# TODO: Make top-level function
+# This is stuff scientists are responsible for!!
+def make_collection(output_dir="./data/", do_frames=True, scale=1, dataset=""):
+    # example dataset name : 3500005820_3
+    # use pandas to load data
+    # a is the full collection!
+    a = pd.read_csv(
+        "//allen/aics/microscopy/EMTImmunostainingResults/EMTTimelapse_7-25-23/Output_CAAX/MigratoryTracksTable_AvgColonyOverlapLessThan0.9_AllPaths.csv"
+    )
+
+    if dataset != "":
+        # convert just the described dataset.
+        plate = dataset.split("_")[0]
+        position = dataset.split("_")[1]
+        c = a.loc[a["Image_Metadata_Plate"] == int(plate)]
+        c = c.loc[c["Image_Metadata_Position"] == int(position)]
+        make_dataset(c, output_dir, dataset, do_frames, scale)
+    else:
+        # for every combination of plate and position, make a dataset
+        b = a.groupby(["Image_Metadata_Plate", "Image_Metadata_Position"])
+        collection = []
+        for name, group in b:
+            dataset = str(name[0]) + "_" + str(name[1])
+            print(dataset)
```

Comment:
nit, but you seem to be using a logger other places, so replace this, or delete?